oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/spi/editor/FulltextDocumentMaker.java

```diff
@@ -336,6 +339,37 @@ private boolean indexProperty(String path,
         return dirty;
     }
 
+    static String getValueAsTruncatedString(Object val) {
+        String value = val.toString();
```

Comment:
Maybe trim the string before testing below for starting and ending with [ ].

Reply:
done